text/0006-shipyard-next-generation.md

```diff
@@ -0,0 +1,317 @@
+    "data": {
+      "deployment": {
+        "strategy": "blue_green",
+        "traffic": "100",
+        "deploymentURI":"https://my-service.domain.com/" // Property added by Keptn-service
+      }
+      "test": {
+
+      }
+    }
+  }
+  ```
+
+
+
+## Breaking changes
+
+This KEP breaks the implementation of Keptn 0.6.0: 
+* Keptn-services (aka. batteries) including contributed services need to be changed to react on a `[task].triggered`, and send a `[task].started` → `[task].finished`.
+* Keptn CloudEvents must be changed.
+* A shipyard migration must take place as outlined below.
+
+### Upgrade path from Shipyard 0.1.2  to 0.2.0
+
+The shipyard specification version 0.1.2 as used by Keptn 0.5.0 and 0.6.0 (last supported Keptn versions) has to be migrated to the new specification. Thus, the following spec: 
+
+```yaml
+stages:
+- name: "dev"
+  deployment_strategy: "direct"
+  test_strategy: "functional"
+- name: "staging"
+  deployment_strategy: "blue_green_service"
+  test_strategy: "performance"
+- name: "production"
+  deployment_strategy: "blue_green_service"
+  remediation_strategy: "automated"
+```
+
+... migrates to:
+
+```yaml
+---
+version: 0.2.0
+kind: Shipyard
+metadata:
+  name: shipyard
+spec:
+  stages:
+  - name: "dev"
+    sequences:
+    - name: artifact-delivery
+      tasks:
+      - name: deployment
+        properties:
+          strategy: direct
+      - name: test
+        properties:
+          kind: functional
+      - name: evaluation 
+      - name: release
+
+  - name: "staging"
+    sequences:
+    - name: artifact-delivery
+      triggers:
+      - dev.artifact-delivery.finished
+      tasks:
+      - name: deployment
+        properties:
+          strategy: blue_green
+      - name: test
+        properties:
+          kind: performance
+      - name: evaluation
+      - name: release
+        
+    - name: rollback
+      triggers:
+      - hardening.artifact-delivery.finished:
+          selector:
+            matchLabels:
+              status: failed
+      tasks:
+      - name: rollback
+    
+  - name: "production"
+    sequences:
+    - name: artifact-delivery 
+      triggers:
+      - hardening.artifact-delivery.finished
+      tasks:
+      - name: deployment
+        properties:
+          strategy: blue_green
+      - name: release
+      
+    - name: remediation
+      triggers: 
+      - production.problem.open            #(smart default)
+      - production.remediation.in-progress #(smart default)
+      tasks:
+      - name: remediation
+      - name: test
+        properties:
+          kind: real_user
+      - name: evaluation
+```
+
+## Ideas for enhancement
+
+- A task can have a property, which defines if the task should be executed depending on previous tasks: 
+  - `Only when all previous tasks have succeeded`
+  - `Even if a previous task has failed`
+  - `Only when a previous taks has failed`
+
+## Open questions
+
+- Do we need a CloudEvent specification for sequence events? 
```

Comment:
I would say yes, in a very basic form like
``sh.keptn.event.${stage}.${sequence}.(triggered|started|finished)``